lib/context.js

```diff
@@ -6,25 +6,40 @@
 'use strict'
 
 const util = require('util')
+const semver = require('semver')
+
+// Version middleware is stable
+// See: https://nextjs.org/docs/advanced-features/middleware
+const MIN_SUPPORTED_VERSION = '12.2.0'
 
 module.exports = function initialize(shim, ctx) {
+  const nextVersion = shim.require('./package.json').version
+
+  if (semver.lt(nextVersion, MIN_SUPPORTED_VERSION)) {
+    shim.logger.debug(
```

Comment:
I wonder if `debug` is the right level of noisy for this notice.

Reply:
i could move to warn